W-17316117 Managed Flex troubleshooting topics (#776)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/flex-troubleshoot-managed-creation.adoc

= Troubleshooting Managed Flex Gateway Creation
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> When creating and deploying a Managed Flex Gateway instance, you might encounter issues. Check these topics for possible causes and solutions.

Reviewer: ```suggestion
When creating and deploying a Managed Flex Gateway, you might encounter issues. Check these topics for possible causes and solutions.
```

* xref:#add-button-disabled[Add Managed Flex Gateway Button is Disabled]
* xref:#private-space[Private Space Not Available]

[[add-button-disabled]]
== Add Managed Flex Gateway Button is Disabled

> The *Add Managed Flex Gateway* button in Runtime Manager could be disabled due to quota limitations or insufficient permissions. This issue can occur when you've reached the maximum number of Managed Gateways allowed for all sizes in your organization, or if you do not have the required _Manage Servers_ and _Read Servers_ permissions in Runtime Manager. 

Reviewer: ```suggestion
The *Add Managed Flex Gateway* button in Runtime Manager might be disabled due to quota limitations or insufficient permissions. This issue occurs if you reach the maximum number of Managed Flex Gateways allowed for all sizes in your organization or if you don't have the required _Manage Servers_ and _Read Servers_ Runtime Manager permissions . 
```

Check Managed Flex Gateway quotas for your organization::

. In Anypoint Platform, go to *Access Management*.
. Click *Runtime Manager* in *Subscription* section of the side navigation panel.
. Review the Managed Gateway quotas to determine if the quota has been exhausted for all Managed Flex Gateway sizes.

> If necessary, you can delete existing Managed Gateways to free up quota. You can also upgrade to a subscription plan with a higher Managed Gateway quota.

Reviewer: ```suggestion
To deploy more Managed Flex Gateways, delete existing gateways or upgrade to a subscription plan with a higher quota.
```

Verify your permissions::

> Check your assigned roles and permissions for Runtime Manager, in Access Management to verify that you have the required _Manage Servers_ and _Read Servers_ permissions.

Reviewer: ```suggestion
In Access Management, ensure that you have the required _Manage Servers_ and _Read Servers_ Runtime Manager permissions.
```

Your Anypoint Platform Admin can add these permissions in Access Management.

> See xref:access-management::configure-teams.adoc#manage-team-permissions[Manage Team Permissions] for more information. 

Reviewer: ```suggestion
For more information, see xref:access-management::configure-teams.adoc#manage-team-permissions[Manage Team Permissions].
```

[[private-space]]
== Private Space Not Available

> If, when creating a new Managed Flex Gateway, you do not see the desired Private Space in the *Deployment Target* list, you may not have the necessary permissions for that Private Space. 

Reviewer: ```suggestion
When creating a new Managed Flex Gateway, if you don't see the desired Private Space in the *Deployment Target* list, you might not have the necessary permissions for that Private Space. 
```

> Verify the Private Space permissions::

Reviewer: ```suggestion
To verify the Private Space permissions:
```

> . In Runtime Manager, access the details of the Private Space you are trying to use.

Reviewer: ```suggestion
. In Runtime Manager, access the details of the Private Space you want to use.
```

. Navigate to the *Environments* tab.
. Confirm that the target organization and environment are within the allowed environments for that private space.

> If necessary, update the Private Space Environments configuration to include the target organization and environment:

Reviewer: ```suggestion
. If it isn't, update the Private Space Environments configuration to include the target organization and environment:
```

Reviewer: I'd combine with the above steps.

> . Modify the Private Space *Environments* configuration to include the target organization and environment.

Reviewer: ```suggestion
.. Modify the Private Space *Environments* configuration to include the target organization and environment.
```

> . Wait a few minutes for the changes to propagate and be reflected in the *Deployment Target* list.

Reviewer: ```suggestion
.. Wait a few minutes for the changes to propagate and be reflected in the *Deployment Target* list.
```
